File: Arcade/Intro/C++/23_boxBlur.cpp
std::vector<std::vector<int>> boxBlur(std::vector<std::vector<int>> image) {
    vector<vector<int>> result(image.size()-2, vector<int>(image[0].size()-2,0));
    for(int i=1;i<image.size()-1;i++){
        for(int j=1;j<image[0].size()-1;j++){
            int sum=image[i-1][j]+image[i-1][j+1]+image[i][j+1]+image[i+1][j+1]+image[i+1][j]+image[i+1][j-1]+image[i][j-1]+image[i-1][j-1]+image[i][j];
            result[i-1][j-1]=sum/9;
            
        }
    }
    return result;
}

File: Arcade/Intro/JavaScript/23_boxBlur.js
function boxBlur(image) {
    var blur = [];
    
    var width = image[0].length;
    var height = image.length;
    
    for (i = 1; i < (height-1); i++) {
        var arr = [];
        for (j = 1; j < (width-1); j++) {
            var sum = image[i-1][j-1] + image[i-1][j] + image[i-1][j+1] +
                image[i][j-1] + image[i][j] + image[i][j+1] +
                image[i+1][j-1] + image[i+1][j] + image[i+1][j+1];
            arr.push(sum/9 - (sum/9 % 1));
        }
        blur.push(arr);
    }
    
    return blur;
}

